src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -466,6 +466,41 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
         }
 
         [Fact]
+        public async Task GetAsync_ResponseWithSetCookieHeaders_AllCookiesRead()
+        {
+            // Cookie names/values, with one of the values deliberately over 128 characters,
+            // which is the size of WinHttpHandler's initial response header buffer size, to
```

Comment:
> Then maybe this test really belongs in the System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs.

I had the same initial thought, but then figured it might be good to have this coverage for both WinHttpHandler and CurlHandler.

I've added some additional tests (via MemberData) and kept it in System.Net.Http for now.